comp/core/telemetry/telemetry.go

```diff
@@ -195,6 +201,14 @@ func (t *telemetryImpl) NewSimpleHistogramWithOpts(subsystem, name, help string,
 		Buckets:   buckets,
 	})}
 
-	_ = t.registry.Register(pc.h)
+	t.registry.MustRegister(pc.h)
 	return pc
 }
+
+func (t *telemetryImpl) GetRegistry() *prometheus.Registry {
```

Comment:
As these methods are only for the mock, it would be great if you can create a `mockTelemetryImpl` struct. 
You can use embeded struct for the implementation. ie
```go
type telemetryImplMock struct {
         telemetryImpl
}
```
This is important because `t.registry` is not thread safe.